core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -2586,7 +2586,7 @@ class ReplicaManager(val config: KafkaConfig,
     // retrieve the UUID here because logManager.handleLogDirFailure handler removes it
     val uuid = logManager.directoryId(dir)
     logManager.handleLogDirFailure(dir)
-    if (dir == config.metadataLogDir) {
+    if (dir == new File(config.metadataLogDir).getAbsolutePath && (zkClient.isEmpty || config.migrationEnabled)) {
```

Comment:
This is not the correct check... you should check `config.processRoles`

Reply:
Addressed in https://github.com/apache/kafka/pull/15262/commits/b28e21a5f96d580e4463e20eb73738a157eb440c